library/iterator.lisp

```diff
@@ -273,32 +216,27 @@ Equivalent to reversing `range-increasing`"
                     (const False)
                     0))
 
-  (declare repeat-forever (:item -> Iterator :item))
-  (define (repeat-forever item)
+  (declare repeat (:item -> Iterator :item))
```

Comment:
can we also add
```
(declare cycle (List :item -> Iterator :item))
```

so that `(cycle (make-list a b c))` produces an iterator `a b c a b c a b c ...`

I don't recommend this but then `(repeat x) == (cycle (make-list x))`